src/submit.py

```diff
@@ -43,6 +46,21 @@ async def _check_git_repo(repo_dir: pathlib.Path):
         click.echo(f"Uncommitted changes in {repo_dir}:")
         click.echo(output)
         click.confirm("Are you sure you want to continue?", abort=True)
+        
+    task_info = await HOOKS.getTask()
+    has_internet_access = bool(task_info.permissions)
+    
+    if has_internet_access:
+        return await _verify_git_repo_pushed(repo_dir)
+    else:
+        click.confirm(
+            "Since this task is running on a container with no internet access, "
```

Comment:
This is a draft message, I assume we either have something better or I should write one (using `viv ssh` I'm assuming, right?)

Reply:
We don't already have a message, writing one is part of completing this issue. `viv ssh` doesn't run commands against the server, only SSHes in, so that wouldn't help . I gave some options [in Slack](https://evals-workspace.slack.com/archives/C07KLBPJ3MG/p1730664144697029?thread_ts=1729695636.683289&cid=C07KLBPJ3MG):

> * [This PR](https://github.com/evals-sandbox/baseline-ops/pull/74) demonstrates the git repo logic. There is now (once that PR is merged) always a git repo at /home/agent , regardless of whether or not the task has internet.
> * When there is no internet, running submit should prompt the user with [instructions to clone the repo to their local computer](https://github.com/poking-agents/headless-human/issues/19#issuecomment-2412430330) and push from there
>   * The git remote details are also demonstrated in baseline-ops. They are currently not saved to the run environment if there is no internet (i.e. no new host in /home/agent/.ssh/config, no remote added to git repo), though that could be changed so that headless-human can give the baseliner the needed instructions to clone and push.
>   * Alternatively, pre-populate the submission repo with a script the baseliner can run to make this easier

Reply:
I assume you mean the baseliner should *copy* the repo to their local machine using `viv scp`?
If not then sorry, I don't know how you mean to clone that. It seems to me like the baseliner doesn't have something like a direct ip of the container that the baseliner is allowed to use

Reply:
Regarding putting the github repo url in the run environment, I assume you mean logic around this:
https://github.com/evals-sandbox/baseline-ops/blob/main/metr/baseline_ops/prep.py#L30
And I'll look if there's some standard way that such info is passed in (I assume we don't want a git remote that just won't work)

Reply:
(and my intuition is to do both of these in a separate PR if that's ok)

Reply:
I hinted at using `viv scp` but didn't add full instructions.

Reply:
No, I mean clone
```
$ tail ~/.ssh/config 
Host baseline
  HostName v0run--1608445550--server
  User agent
  ForwardAgent Yes
  ProxyJump ssh-user@147.185.40.136:30022
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no

$ git clone ssh://baseline/home/agent baseline
Cloning into 'baseline'...
Warning: Permanently added 'v0run--1608445550--server' (ED25519) to the list of known hosts.
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 14 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (14/14), 11.43 KiB | 300.00 KiB/s, done.

$ cd baseline && ls -la

$ ls -la
total 48
drwxr-xr-x  10 samijawhar  staff   320 Nov  7 11:29 .
drwxr-x---+ 38 samijawhar  staff  1216 Nov  7 11:29 ..
drwxr-xr-x  12 samijawhar  staff   384 Nov  7 11:29 .git
-rw-r--r--   1 samijawhar  staff    98 Nov  7 11:29 .gitignore
drwxr-xr-x   3 samijawhar  staff    96 Nov  7 11:29 __pycache__
-rw-r--r--   1 samijawhar  staff   193 Nov  7 11:29 api_key.txt
drwxr-xr-x   3 samijawhar  staff    96 Nov  7 11:29 data
-rw-r--r--   1 samijawhar  staff  5373 Nov  7 11:29 rust_codecontests_utils.py
-rw-r--r--   1 samijawhar  staff  7455 Nov  7 11:29 score.py
drwxr-xr-x   4 samijawhar  staff   128 Nov  7 11:29 solution
```

Reply:
Huh? How did `~/.ssh/config` get that `baseline` entry? Is this supposed to happen on the baseliner's local machine?

Reply:
I created it. The baseliner is provided with the info to connect over SSH. Currently it's provided as an SSH command `ssh -J ...`, but it's the same info.

Reply:
Updated the message to stop recommending scp meanwhile.

Reply:
Sami [said](https://github.com/poking-agents/headless-human/pull/29#discussion_r1835231847):
> Need complete, explicit instructions here. The baseliner should NOT be prompted to ask for help, that's a big blocker for them and bottleneck for us.

Reply:
I prefer you tell me much more on how it's provided so that I can write reasonable documentation. I don't think I've seen this `ssh -J` command anywhere and I haven't done the full baseliner process (didn't see their welcome messages and so on).

Reply:
Updated the comment based on my current understanding. I am very unsure if this is enough or if I'm more things. I definitely don't feel confident about it